File: connect.js
const {Builder, By, until} = require("selenium-webdriver");
const chrome = require("selenium-webdriver/chrome");
const BasePage = require("./src");

async function sleep(timeInS) {
    await new Promise((resolve) => setTimeout(resolve, timeInS * 1000))
}

async function startBot() {
    const page = new BasePage();
    let site = "https://www.linkedin.com";
    await page.visit(site);

    await sleep(5)

    await page.signin();
    await sleep(35)
    var count = 0 ; 

    for( let i = 2 ; i < process.argv.length ; i++ )
    {
        topic = process.argv[i] ; 
        let num = 1 ; 
        site = "https://www.linkedin.com/search/results/people/?keywords=" + topic + "&origin=GLOBAL_SEARCH_HEADER&page=" + num ;
        // &origin=GLOBAL_SEARCH_HEADER&sid=wW%3A
        // &origin=CLUSTER_EXPANSION&sid=J*E
        page.visit(site);
        await sleep(5)
        
        for( num = 1 ; num < 10 ; num++ )
        {
            site = "https://www.linkedin.com/search/results/people/?keywords=" + topic + "&origin=GLOBAL_SEARCH_HEADER&page=" + num ;

            await page.visit(site);
            for (let i = 3; i < 10000; i++) {
                if (i % 10 === 9) {
                    page.visit(site);
                    await sleep(8)
                    page.scrollToBottom();
                    await sleep(2)
                }
                try {

                    // let button = await this.findByClassName("artdeco-button artdeco-button--2 artdeco-button--secondary"); // Replace "your-button-class-name" with the actual class name of the button
                    // let buttonId = await button.getAttribute("id"); // Get the id attribute of the button
                    // let ariaLabel = await buttonId.getAttribute("aria-label"); // Get the aria-label attribute of the button
                  
                    // if ( ariaLabel.includes("Invite") )
                    // {
                    //     await page.pressConnectButton();
                    //     await page.pressConnectButton2();
                    //     count++ ; 
                    // }
                    // else 
                    // {
                    //     await page.pressFollowButton();
                    //     count++ ; 
                    // }

                    await page.pressConnectButton() ; 
                    await new Promise((resolve) => setTimeout(resolve, 3000))
                    page.scrollToBottom();
                    await sleep(3)
                } catch (e) {
                    console.error(e)
                }
            }
            await new Promise((resolve) => setTimeout(resolve, 5000))
        }

    }

    console.log("Total buttons clicked : " , count ) ; 
}
(async () => {
    if( process.argv.length > 2 ) 
        await startBot()
    else
    {
        console.log("No topics Entered!!!") ; 
        process.exit(0) ; 
    } 

})()


    /* 
    ALGORITHM 
    Search input topics --> input from the user in bash terminal
    if( input = 1 ) 
        single string i.e. site 
    else 
        SiteGenr 
    Click on "see all people results" 
    loop( findPages() )
    {
        Only click on follow / connect buttons  
        if( connect_button ) 
        {
            either 
                click( send without a note ) 
                await Promise 2 seconds
            or 
                click( Add a note ) 
                Fill the placeholder with message 
                click( send ) 
                await Promise 7 seconds 
        }
        scrollToBottom()
        clickNextPage() 
    }

    */

    /* 
    
    Search input topics --> input from the user in bash terminal
    if( input = 1 ) 
        single string i.e. site 
    else 
        Sites[] = SiteGenr( ...inputs )  
    Sites.forEach( site => 
    {
        
        Click on "see all people results" 
        loop( findPages() )
        {
            Only click on follow / connect buttons  
            if( connect_button ) 
            {
                either 
                    click( send without a note ) 
                    await Promise 2 seconds
                or 
                    click( Add a note ) 
                    Fill the placeholder with message 
                    click( send ) 
                    await Promise 7 seconds 
            }
            scrollToBottom()
            clickNextPage() 
        }

    })

*/

File: connect.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException
import sys
import time
import curses
import random

# # Live count Animation with color
# stdscr = curses.initscr()
# curses.curs_set(0)  
# stdscr.nodelay(1)  
# stdscr.timeout(100)
# colors = [curses.COLOR_RED, curses.COLOR_GREEN, curses.COLOR_BLUE, curses.COLOR_YELLOW]

if( len( sys.argv ) < 2  ) : 
    print("No topic Declared!!!") 
    exit(1) 

# linux path : executable_path='/usr/bin/chromedriver'
# windows path : executable_path='/path/to/your/chromedriver/chromedriver.exe'
service = webdriver.ChromeService(executable_path='/usr/bin/chromedriver')
driver = webdriver.Chrome(service=service)

driver.get('https://www.linkedin.com')
time.sleep(2)

username = driver.find_element(By.NAME, "session_key")
password = driver.find_element(By.NAME, "session_password")

username.send_keys('Enter your email here')
password.send_keys('Enter your password here')
time.sleep(2)

submit = driver.find_element(By.XPATH, "//button[@type='submit']").click()
time.sleep(35)

count = 0

try: 
    for i in range( 1 , len( sys.argv ) ) :
        for num in range( 11 , 100 ) : 
            site = "https://www.linkedin.com/search/results/people/?keywords=" + sys.argv[i] + "&origin=GLOBAL_SEARCH_HEADER&page=" + str(num)
            driver.get(site) 
            time.sleep(2)

            all_buttons = driver.find_elements(By.TAG_NAME, "button")
            connect_buttons = [btn for btn in all_buttons if ( btn.text == "Connect" ) ]

            for btn in connect_buttons:

                # stdscr.clear()
                # stdscr.addstr(0, 0, "Live Count: {}".format(count), curses.color_pair(random.choice(colors)))
                # stdscr.refresh()

                driver.execute_script("arguments[0].click();", btn)
                time.sleep(2)

                if( sys.argv[ len(sys.argv) - 1 ] == '0' ) :  
                    # # SEND WITHOUT NOTE 
                    send = driver.find_element(By.XPATH, "//button[@aria-label='Send now']")
                    driver.execute_script("arguments[0].click();", send)
                    time.sleep(2) 
                    print("Connections made : " , count) 

                    # # Finding connection name
                    # name = driver.find_element(By.TAG_NAME, "strong").text
                    # print( name + "\n" ) 
                    # print(f"Connected to {name} \n")

                else : 
                    # # SEND WITH NOTE 
                    sendWithNote = driver.find_element(By.XPATH, "//button[@aria-label='Add a note']")
                    driver.execute_script("arguments[0].click();", sendWithNote)
                    time.sleep(3) 

                    # Finding person name
                    name = driver.find_element(By.TAG_NAME, "strong").text
                    print( name + "\n" ) 

                    # Paste message 
                    inputBox = driver.find_element(By.ID, "custom-message")
                    input_text = f"Hi {name}, Please allow me to be a part of your connection family"
                    inputBox.send_keys(input_text)
                    time.sleep(3)

                    # Final send 
                    send = driver.find_element(By.XPATH, "//button[@aria-label='Send now']")
                    driver.execute_script("arguments[0].click();", send)

                    print(f"clicked on {name} \n")


                count += 1 
                close = driver.find_element(By.XPATH, "//button[@aria-label='Dismiss']")
                driver.execute_script("arguments[0].click();", close)
                time.sleep(2)

                

                if( count == 100 ) : 
                    print("Made 100 Connections - Come again after a week")
                    exit(1) 

            print(f"Made {count} connections till page-{num}")
            print(f"-------------TURNING INTO PAGE : {num}-------------")

except (KeyboardInterrupt, NoSuchElementException ):
    print("\n")
    print(f"Successfully accomplished {count} connections")
    print(f"You'll get banned on LinkedIn if you exceed {100-count} more connections this week")
    exit(1) 
else : 
    print(f"Successfully accomplished {count} connections")
    print(f"You'll get banned on LinkedIn if you exceed {100-count} more connections this week")